app/pages/_app.js

```diff
@@ -29,13 +30,31 @@ const GlobalStyle = createGlobalStyle`
   html {
     scroll-behavior: smooth;
   }
+
+  html.normal-scroll {
+    scroll-behavior: auto;
+  }
 `;
 
 class App extends PureComponent {
+  // scroll-behaviour: smooth on html breaks nextjs linking, will not go to top
+  // of page when routing. Switching to normal scroll between routing fixes this
+  // See https://github.com/vercel/next.js/issues/20125
+  componentDidMount() {
```

Comment:
we might need to de-attach the event handlers in `componentWillUnmount` middleware?
https://nextjs.org/docs/api-reference/next/router#usage-6